src/psyclone/psyir/transformations/region_trans.py

```diff
@@ -166,10 +167,22 @@ def validate(self, nodes, options=None):
         # Check that the proposed region contains only supported node types
         if options.get("node-type-check", True):
             for child in node_list:
-                if child.walk(self.excluded_node_types):
-                    raise TransformationError(
-                        f"Nodes of type '{self.excluded_node_types}' cannot "
-                        f"be enclosed by a {self.name} transformation")
+                for bad in child.walk(self.excluded_node_types):
+                    # Ideally we'd just use `wrong.debug_string()` always
+                    # but this sometimes causes errors if nodes do their
+                    # own validation (rather than relying on the backend),
```

Comment:
Yes, it can make life difficult. However, there are certain conditions that we cannot check for until we're confident that the user has finished doing transformations and that means waiting until code-generation.

Reply:
I was replying to this conversation, but my comment turned quite long, so I put it #2680 instead.